website/blog/2025-01-20-podman-quadlet.md

```diff
@@ -0,0 +1,206 @@
+
+What’s a more lightweight solution for running autonomous applications with multiple interacting containers?
+
+In this guide, we'll dive into what Quadlets are, their benefits, and how to utilize them within Podman Desktop.
+
+<!-- truncate -->
+
+## What Are Podman Quadlets?
+
+Podman Quadlets allow you to manage containers declaratively using systemd.
+Since version 4.4, Podman can create, start, and manage containers (including pulling images, creating volumes, and managing pods) through systemd.
+
+Quadlets are simplified configuration files—recognized by their specific extensions,
+such as `*.container`, `*.pod`, or `*.image` that systemd processes during startup or when you reload the daemon (systemctl daemon-reload).
+
+Quadlets generate the equivalent systemd unit files, streamlining the container management process.
+
+### Why Use Quadlets?
+
+- **Declarative Configuration**: Similar to Compose or Kubernetes manifests, Quadlets allow you to declare what you want to run, simplifying the workload setup.
+- **Tight System Integration**: Quadlets align with Podman’s philosophy of integrating seamlessly with Linux, leveraging systemd’s process management capabilities.
+- **Ease of Automation**: Quadlets make it simple to configure containers to start at boot, restart on failure, and more.
+
+### Example: A Quadlet File for Nginx
+
+Below is an example of an `nginx.container` Quadlet file, which starts an nginx container at boot:
+
+```editorconfig title="~/.config/containers/systemd/nginx.container"
+# nginx.container
+[Container]
+ContainerName=nginx
+Image=nginx
+PublishPort=80:8080
+
+[Service]
+Restart=always
+```
+
+This configuration ensures the container restarts automatically if stopped, and exposes port 8080.
+
+---
+
+## Using the Podman Quadlet Extension in Podman Desktop
+
+Managing Quadlets directly on non-Linux platforms can be challenging due to virtualized environments (e.g., WSL or Hyper-V).
+Fortunately, the Podman Desktop extension Podman Quadlet simplifies this process, enabling you to list, generate, and edit Quadlets visually.
+
+### Key Features of the Extension
+
+- **Integration with Podlet**: Generates Quadlets from existing Podman objects.
+- **Quadlet Management UI**: Provides a dedicated interface to list, edit, delete, start, and stop Quadlets.
+- **Logs Viewer**: Fetches and displays systemd logs using journalctl for troubleshooting.
+
+### Installation
+
+If you already have the latest version of Podman Desktop, you can <a href="podman-desktop:extension/podman-desktop.quadlet">**click here to install the Podman Quadlet extension**</a>.
+
+Alternatively, navigate to the Extensions page within Podman Desktop to install it
+
+### Listing Quadlets :clipboard:
+
+On the Podman Quadlet page, you can view all the Quadlets available across your Podman machines. To update the list, click **Refresh**.
```

Comment:
Can we use a screenshot in which we can see a few quadlets? Right now, it shows no quadlets found on the system.

Reply:
Yes, sure, this would be way better!